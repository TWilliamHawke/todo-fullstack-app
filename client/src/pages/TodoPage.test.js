import { shallow } from 'enzyme'
import ConnectedTodoPage, { TodoPage } from './TodoPage'
import React from 'react'
import configureMockStore  from 'redux-mock-store'
import {shallowToJson} from 'enzyme-to-json'

describe('dummy component', () => {
  it('should rendered correctly', () => {
    const wrapper = shallow(<TodoPage errors='error array' />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should render errorList with correct data', () => {
    const wrapper = shallow(<TodoPage errors='error array' />)
    expect(wrapper.find('ErrorList').props().data).toBe('error array')
  })
})

describe('connected component', () => {
  let store, wrapper

  beforeAll(() => {
    const mockStore = configureMockStore()
    store = mockStore({todo: {errors: 'errors array'}})
    wrapper = shallow(<ConnectedTodoPage store={store} />).find(TodoPage)
  })

  it('should receive props from store', () => {
    expect(wrapper.props().errors).toBe('errors array')
  })

})