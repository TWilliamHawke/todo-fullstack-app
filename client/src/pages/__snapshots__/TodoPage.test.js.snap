// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dummy component should rendered correctly 1`] = `
<Fragment>
  <Connect(AppHeader) />
  <div
    className="top-panel d-flex"
  >
    <Connect(SearchPanel) />
    <Connect(ItemStatusFilter) />
  </div>
  <Connect(TodoList) />
  <ErrorList
    data="error array"
  />
  <Connect(AddPanel) />
  <Connect(LoadMoreButton) />
</Fragment>
`;
